experiments/compositions/composition/tests/testclient/client.go

```diff
@@ -219,6 +238,118 @@ func (c *Client) MustExist(objs []*unstructured.Unstructured, timeout time.Durat
 	}
 }
 
+// MustHaveCondition - validate if the .status.conditions[] has the condition
+func (c *Client) MustHaveCondition(obj *unstructured.Unstructured, condition *metav1.Condition, timeout time.Duration) {
+	c.T.Helper()
+
+	toMatch := ExtractGVKNN(obj)
+	c.T.Logf("Checking condition %q present in object %q on cluster %q", condition.Type, toMatch, c)
+
+	unmatched := []*unstructured.Unstructured{obj}
+	errMissing := fmt.Errorf("condition missing")
+	errMismatch := fmt.Errorf("condition mismatch")
+
+	retryFrequency := getFrequency(c.T, opDuration, timeout)
+	err := retry.OnError(retryFrequency, func(err error) bool {
+		return apierrors.IsNotFound(err) || errors.Is(err, errMissing) || errors.Is(err, errMismatch)
+	}, func() (err error) {
+		match := func(obj *unstructured.Unstructured) error {
+			read, err := c.Read(obj)
+			if err != nil {
+				return err
+			}
+			conditions, ok, err := unstructured.NestedSlice(read.Object, "status", "conditions")
+			if err != nil {
+				return err
+			}
+			if !ok {
+				return fmt.Errorf("%w: .status.conditions not found", errMissing)
+			}
+			found := false
+			for i := range conditions {
+				cond, ok := conditions[i].(map[string]interface{})
+				if !ok {
+					return fmt.Errorf("%w: .condition[%d] not of type map[string]interface{}", errMissing, i)
```

Comment:
Similarly, printing the actual type would probably be more useful than `%w` here

Reply:
Iam not sure if we use %T the errType would be propagated. Will try.